src/Domain/User.php

```diff
@@ -10,94 +10,88 @@
 /**
  * Domain model for User within the Baikal system
  */
-final class User
-{
+final class User extends DomainObject {
+
     /**
-     * @var Username
+     * Unique ID
+     *
+     * @var mixed
      */
-    private $username;
+    public $id;
```

Comment:
Less typing I suppose ;) This comment ties into your other comment though. If the object is allowed to have an invalid/incomplete state, making the properties public is not a big deal, because we're doing validation a bit later anyway.